### PR TITLE
docs: add eval documentation

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/user-guide/evaluation.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+A detailed description of running evaluation with ``.nemo`` models can be found in Eleuther AI's `documentation <https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file#nvidia-nemo-models>`_.
+Single- and multi-GPU evaluation is supported. The following is an example of running evaluation on the ``hellaswag``, ``super_glue``, and ``winogrande`` tasks using a ``.nemo`` file from NeMo-Aligner.
+Note that it is recommended, but not required, to unzip your ``.nemo`` file prior to running evaluations.
+
+.. code-block:: bash
+
+   mkdir unzipped_checkpoint
+   tar -xvf /path/to/model.nemo -c unzipped_checkpoint
+
+   PYTHONPATH=/path/to/lm-evaluation-harness:${PYTHONPATH} torchrun --nproc-per-node=8 --no-python  lm_eval --model nemo_lm \
```
Comment (terrykong) [inline docs/user-guide/evaluation.rst:36]: is updating the pythonpath necessary if you pip install it?

Comment (ashors1) [reply docs/user-guide/evaluation.rst:36]: You're right, it shouldn't be necessary. I'll update

File: docs/user-guide/evaluation.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
+.. include:: /content/nemo.rsts
+
+.. _nemo-aligner-eval:
+
+Evaluating a Trained Model
```
Comment (jgerh) [inline docs/user-guide/evaluation.rst:5]: revise title to an imperative verb

Evaluate a Trained Model
@@@@@@@@@@@@

File: docs/user-guide/evaluation.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+After training a model, you may want to run evaluation to understand how the model performs on unseen tasks. You can use Eleuther AI's `Language Model Evaluation Harness <https://github.com/EleutherAI/lm-evaluation-harness>`_
+to quickly run a variety of popular benchmarks, including MMLU, SuperGLUE, HellaSwag, and WinoGrande.
+A full list of supported tasks can be found `here <https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/tasks/README.md>`_.
+
+Install LM Evaluation Harness
+#############################
```
Comment (jgerh) [inline docs/user-guide/evaluation.rst:13]: revist title

Install the LM Evaluation Harness
#########################

File: docs/user-guide/evaluation.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+   cd lm-evaluation-harness
+   pip install -e .
+
+
+Run Evaluations
+###############
+
+A detailed description of running evaluation with ``.nemo`` models can be found in Eleuther AI's `documentation <https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file#nvidia-nemo-models>`_.
+Single- and multi-GPU evaluation is supported. The following is an example of running evaluation on the ``hellaswag``, ``super_glue``, and ``winogrande`` tasks using a ``.nemo`` file from NeMo-Aligner.
+Note that it is recommended, but not required, to unzip your ``.nemo`` file prior to running evaluations.
```
Comment (jgerh) [inline docs/user-guide/evaluation.rst:29]: suggested revision

Please note that while it is recommended, you are not required to unzip your .nemo file before running evaluations.

File: docs/user-guide/evaluation.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+A detailed description of running evaluation with ``.nemo`` models can be found in Eleuther AI's `documentation <https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file#nvidia-nemo-models>`_.
+Single- and multi-GPU evaluation is supported. The following is an example of running evaluation on the ``hellaswag``, ``super_glue``, and ``winogrande`` tasks using a ``.nemo`` file from NeMo-Aligner.
+Please note that while it is recommended, you are not required to unzip your .nemo file before running evaluations.
+
+.. code-block:: bash
+
+   mkdir unzipped_checkpoint
+   tar -xvf /path/to/model.nemo -c unzipped_checkpoint
+
+   torchrun --nproc-per-node=8 --no-python  lm_eval --model nemo_lm \
```
Comment (terrykong) [inline docs/user-guide/evaluation.rst:36]: nit, two spaces between no-python and lm_eval

File: docs/user-guide/evaluation.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+   git clone --depth 1 https://github.com/EleutherAI/lm-evaluation-harness
+   cd lm-evaluation-harness
+   pip install -e .
+
+
+Run Evaluations
+###############
+
+A detailed description of running evaluation with ``.nemo`` models can be found in Eleuther AI's `documentation <https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file#nvidia-nemo-models>`_.
+Single- and multi-GPU evaluation is supported. The following is an example of running evaluation on the ``hellaswag``, ``super_glue``, and ``winogrande`` tasks using a ``.nemo`` file from NeMo-Aligner.
```
Comment (terrykong) [inline docs/user-guide/evaluation.rst:28]: maybe include the detail that this example requires 8 gpus

**Pull request merged**